src/app/contactmanager/services/user.service.ts

```diff
@@ -8,38 +8,30 @@ import { BehaviorSubject, Observable } from 'rxjs';
 })
 export class UserService {
 
-  private _users = new BehaviorSubject<User[]>([]);
-  private dataStore: { users: User[]; } = { users: [] };
+  users = signal<User[]>([]);
 
   http = inject(HttpClient);
 
-  get users(): Observable<User[]> {
-    return this._users.asObservable();
-  }
-
   addUser(user: User): User {
-    user.id = this.dataStore.users.length + 1;
-    this.dataStore.users.push(user);
-    this._users.next(this.dataStore.users);
-
+    user.id = this.users().length + 1;
+    this.users.update(u => [...u, user]);
     return user;
   }
 
   userById(id: number) {
-    return this.dataStore.users.find(x => x.id == id);
+    return this.users().find(x => x.id == id);
   }
 
   loadAll() {
```

Comment:
Another way to handle this would be to move the contents of the ` loadAll ` method into the constructor. That way there is no need to trigger loading the users in the side nav component onInit method. 

This is an example 
https://github.com/lara-newsom/intro-to-angular/blob/main/src/app/services/product-signals.service.ts#L33

Reply:
Smart, did it :-)